app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,54 @@
+class CommentsController < ApplicationController
+  before_action :set_repository
+  respond_to :js, :html
+
+  # POST /repositories/1/comments
+  # POST /repositories/1/comments.json
+  def create
+    @comment = @repository.comments.new(comment_params)
+    @comment.author = current_user
+    authorize @comment
+
+    if @comment.save
+      @repository.create_activity :create_comment,
+                                  owner:      current_user,
+                                  parameters: {
+                                    comment_id:     @comment.id,
+                                    comment_title:  @comment.title,
+                                    comment_body:   @comment.body,
+                                    comment_create: @comment.created_at
+                                  }
+      respond_with(@comment)
+    else
+      respond_with @comment.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /repositories/1/comments/1
+  # DELETE /repositories/1/comments/1.json
+  def destroy
+    @comment = @repository.comments.find(params[:id])
+    authorize @comment
+
+    @repository.create_activity :destroy_comment,
+                                owner:      current_user,
+                                parameters: {
+                                  comment_id:     @comment.id,
+                                  comment_title:  @comment.title,
+                                  comment_body:   @comment.body,
+                                  comment_create: @comment.created_at
+                                }
+    @comment.destroy
+    respond_with @comment
+  end
+
+  private
+
+  def set_repository
+    @repository = Repository.find(params[:repository_id])
+  end
+
+  def comment_params
```

Comment:
If you remove the `:title` attribute, this method won't be necessary anymore I think.